source/apickli/apickli-gherkin.js

```diff
@@ -272,3 +272,9 @@ Then(/^value of scenario variable (.*) should be (.*)$/, function(variableName,
     callback(new Error('value of variable ' + variableName + ' isn\'t equal to ' + variableValue));
   }
 });
+
+When(/^I clear the request$/, function(callback) {
```

Comment:
@rparthas - you are only clearing `httpRequestOptions` and `requestBody` - what about the other properties that may or may not have been populated earlier in the feature, e.g. `queryParameters`.

A cleaner approach might be to reset the entire Apickli object that you create during the cucumber's `Before` event handler. However that cannot be done in `apickli-gherkin` like this as the implementation might depend on many external configuration, therefore cannot be implemented for everyone - but it is a very good candidate to be implemented in your own custom step_definitions file.